### PR TITLE
Convert to V2 addon

### PR DESCRIPTION
**Pull request by bwbuchanan**
This is an attempt at converting `ember-load-initializers` to the v2 addon format. Feedback welcomed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ language: node_js
 node_js:
```
Comment (SergeAstapov) [inline .travis.yml:3]: @bwbuchanan maybe just delete this file and provide GitHub actions config instead?

**Comment by SergeAstapov** [comment]
@bwbuchanan would you like to incorporate recommendations from https://github.com/embroider-build/embroider/blob/main/PORTING-ADDONS-TO-V2.md?

**Comment by NullVoxPopuli** [comment]
@SergeAstapov is this something we should take over? :sweat_smile:

**Comment by chancancode** [comment]
I suppose it doesn’t hurt, but does it really matter that this is v1 or v2 when the core thing that this addon does is iterating over require.entries to look for the initializers? Shouldn’t this functionality be implemented in embroider natively and make this addon unnecessary/no-op under embroider?

**Comment by SergeAstapov** [comment]
> @SergeAstapov is this something we should take over? 😅

@NullVoxPopuli I'd love gets this over the finish line! although this would fall somewhere in the queue of addons I do have maintainer access.
I can seek help of co-workers to get this over the finish line.
We just need someone who can land PRs and do the release.

> I suppose it doesn’t hurt, but does it really matter that this is v1 or v2 when the core thing that this addon does is iterating over require.entries to look for the initializers?

@chancancode as this point in time, the main problem with this addon is that fact that it depends on `ember-cli-typescript@2`, which is problematic per https://github.com/ember-cli/ember-load-initializers/issues/291#issuecomment-1285358767. So at least maintenance like bumping that to v5 is overdue, which kinda brings a question if we should do one step over and convert to v2 which is not that hard then.

> Shouldn’t this functionality be implemented in embroider natively and make this addon unnecessary/no-op under embroider?

TBH it always was a mystery to me why this addon even exists (maybe was a part of migration story from ember-app-kit to Ember CLI?).
having this baked into Embroider IMO sounds like a logical step long term.

**Comment by chancancode** [comment]
Yeah, if it's close and easy, like I said, I don't think it hurts to keep things up-to-date here. But personally, I think if this addon is still needed with Embroider/Polaris, then there is something deeply wrong. (Same deal as the fingerprinting stuff, etc.) So, personally, I think the priority should be making sure the functionality is implemented natively in Embroider.

**Comment by ef4** [comment]
I think it's fine to make this change, and separately after we do https://github.com/emberjs/rfcs/blob/strict-es-module-support/text/0938-strict-es-module-support.md, use that to eliminate the dependence on requirejs.entries.

**Comment by ef4** [comment]
Closing in favor of #306. 

In summary: 
 - this library does want to continue to exist for anybody who depends on addons that contain initializers that expect to run.   - It has been extended to accept the set of modules explicitly as an argument, eliminating the requirement that you're using AMD. 
 - Instead of converting from v1 to v2 addon, it can just be a plain NPM library, because it doesn't need any ember-specific stuff.

**Pull request closed without merging**